### PR TITLE
const_evaluatable: expand abstract consts in try_unify

### PR DESCRIPTION
**Pull request by BoxyUwU**
See this [zulip topic](https://rust-lang.zulipchat.com/#narrow/stream/260443-project-const-generics/topic/combining.20const.20bounds) for more info

cc @lcnr 
r? @oli-obk

### CODE REVIEW

File: compiler/rustc_trait_selection/src/traits/const_evaluatable.rs [hunk 1]
```diff
@@ -609,9 +609,27 @@ where
 /// Tries to unify two abstract constants using structural equality.
 pub(super) fn try_unify<'tcx>(
     tcx: TyCtxt<'tcx>,
-    a: AbstractConst<'tcx>,
-    b: AbstractConst<'tcx>,
+    mut a: AbstractConst<'tcx>,
+    mut b: AbstractConst<'tcx>,
 ) -> bool {
+    while let Node::Leaf(a_ct) = a.root() {
```
Comment (oli-obk) [inline compiler/rustc_trait_selection/src/traits/const_evaluatable.rs:615]: Leave a comment here that this is essentially performing generic substitution, but on the fly.

File: src/test/ui/const-generics/const_evaluatable_checked/nested-abstract-consts-1.rs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+// run-pass
```
Comment (oli-obk) [inline src/test/ui/const-generics/const_evaluatable_checked/nested-abstract-consts-1.rs:1]: Please leave a comment in each of the tests giving a quick human readable explanation of what the test is testing

**Review by oli-obk — CHANGES_REQUESTED** [review]

**Comment by oli-obk** [comment]
Holy :cow: thanks! This was fast and is much better than I anticipated.

File: compiler/rustc_trait_selection/src/traits/const_evaluatable.rs (hunk 1 above)
Comment (BoxyUwU) [reply compiler/rustc_trait_selection/src/traits/const_evaluatable.rs:618]: I didn't write a comment saying *exactly* that but there's now a comment, hope that's fine too ^^

**Comment by oli-obk** [comment]
@bors r-

tidy is unhappy. You can run it locally with `./x.py test src/tools/tidy`

**Pull request merged**